subgraphs/seaport/src/mapping.ts

```diff
@@ -0,0 +1,691 @@
+    marketplace.cumulativeUniqueTraders = 0;
+    marketplace.save();
+  }
+  return marketplace;
+}
+
+function getOrCreateCollectionDailySnapshot(
+  collection: string,
+  timestamp: BigInt
+): CollectionDailySnapshot {
+  let snapshotID = collection
+    .concat("-")
+    .concat((timestamp.toI32() / SECONDS_PER_DAY).toString());
+  let snapshot = CollectionDailySnapshot.load(snapshotID);
+  if (!snapshot) {
+    snapshot = new CollectionDailySnapshot(snapshotID);
+    snapshot.collection = collection;
+    snapshot.blockNumber = BIGINT_ZERO;
+    snapshot.timestamp = BIGINT_ZERO;
+    snapshot.royaltyFee = BIGDECIMAL_ZERO;
+    snapshot.dailyMinSalePrice = BIGDECIMAL_MAX;
+    snapshot.dailyMaxSalePrice = BIGDECIMAL_ZERO;
+    snapshot.cumulativeTradeVolumeETH = BIGDECIMAL_ZERO;
+    snapshot.dailyTradeVolumeETH = BIGDECIMAL_ZERO;
+    snapshot.marketplaceRevenueETH = BIGDECIMAL_ZERO;
+    snapshot.creatorRevenueETH = BIGDECIMAL_ZERO;
+    snapshot.totalRevenueETH = BIGDECIMAL_ZERO;
+    snapshot.tradeCount = 0;
+    snapshot.dailyTradedItemCount = 0;
+    snapshot.save();
+  }
+  return snapshot;
+}
+
+function getOrCreateMarketplaceDailySnapshot(
+  timestamp: BigInt
+): MarketplaceDailySnapshot {
+  let snapshotID = (timestamp.toI32() / SECONDS_PER_DAY).toString();
+  let snapshot = MarketplaceDailySnapshot.load(snapshotID);
+  if (!snapshot) {
+    snapshot = new MarketplaceDailySnapshot(snapshotID);
+    snapshot.marketplace = EXCHANGE_ADDRESS.toHexString();
+    snapshot.blockNumber = BIGINT_ZERO;
+    snapshot.timestamp = BIGINT_ZERO;
+    snapshot.collectionCount = 0;
+    snapshot.cumulativeTradeVolumeETH = BIGDECIMAL_ZERO;
+    snapshot.marketplaceRevenueETH = BIGDECIMAL_ZERO;
+    snapshot.creatorRevenueETH = BIGDECIMAL_ZERO;
+    snapshot.totalRevenueETH = BIGDECIMAL_ZERO;
+    snapshot.tradeCount = 0;
+    snapshot.cumulativeUniqueTraders = 0;
+    snapshot.dailyTradedItemCount = 0;
+    snapshot.dailyActiveTraders = 0;
+    snapshot.dailyTradedCollectionCount = 0;
+    snapshot.save();
+  }
+  return snapshot;
+}
+
+function getNftStandard(collectionID: string): string {
+  let erc165 = ERC165.bind(Address.fromString(collectionID));
+
+  let isERC721Result = erc165.try_supportsInterface(
+    Bytes.fromHexString(ERC721_INTERFACE_IDENTIFIER)
+  );
+  if (isERC721Result.reverted) {
+    log.warning("[getNftStandard] isERC721 reverted on {}", [collectionID]);
+  } else {
+    if (isERC721Result.value) {
+      return NftStandard.ERC721;
+    }
+  }
+
+  let isERC1155Result = erc165.try_supportsInterface(
+    Bytes.fromHexString(ERC1155_INTERFACE_IDENTIFIER)
+  );
+  if (isERC1155Result.reverted) {
+    log.warning("[getNftStandard] isERC1155 reverted on {}", [collectionID]);
+  } else {
+    if (isERC1155Result.value) {
+      return NftStandard.ERC1155;
+    }
+  }
+
+  return NftStandard.UNKNOWN;
+}
+
+// Use best effort to figure out the following data that construct a `Sale`:
```

Comment:
👍 doing the best with what we have

I'm wondering does would Dune queries for Seaport be helpful for navigating through buyer/seller or fees (I think these are the more ambiguous fields to resolve)? 
ref: https://dune.com/Marcov/opensea-seaport

Reply:
not sure exactly how they distinguish between types of fees or between buyer/seller

Reply:
yeah tbh i have no clue what they do under the hood

https://dune.com/queries/911780/1594223

this table `seaport."view_transactions"` has both `buyer` and `seller` column, Dune did some thing to figure them out before giving the data to their users